packages/eslint-plugin-excel-custom-functions/src/rules/no-office-read-calls.ts

```diff
@@ -0,0 +1,178 @@
+import { TSESTree, ESLintUtils } from "@typescript-eslint/experimental-utils";
+import { getCustomFunction, isOfficeObject, isOfficeFuncWriteOrRead, OfficeCalls, getFunctionStarts2, isHelperFunc, bubbleUpNewCallingFuncs, getFunctionDeclarations, superNodeMe } from './utils'
+import { RuleContext, RuleMetaDataDocs, RuleMetaData  } from '@typescript-eslint/experimental-utils/dist/ts-eslint';
+import ts from 'typescript';
+
+/**
+ * @fileoverview Prevents office api calls
+ * @author Artur Tarasenko (artarase)
+ */
+"use strict";
+
+const createRule = ESLintUtils.RuleCreator(
+  () => 'https://github.com/OfficeDev/Office-Addin-Scripts',
+);
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+type Options = unknown[];
+type MessageIds = 'officeReadCall';
+
+export = {
+    name: 'no-office-read-calls',
+
+    meta: {
+        docs: {
+            description: "Prevents office api calls",
+            category: <RuleMetaDataDocs["category"]>"Best Practices",
+            recommended: <RuleMetaDataDocs["recommended"]>"warn",
+            requiresTypeChecking: true,
+            url: 'https://github.com/OfficeDev/Office-Addin-Scripts'
+        },
+        type: <RuleMetaData<MessageIds>["type"]> "problem",
+        messages: <Record<MessageIds, string>> {
+            officeReadCall: "No Office API read calls within Custom Functions"
+        },
+        schema: []
+    },
+
+    create: function(ruleContext: RuleContext<MessageIds, Options>): {
+        CallExpression: (node: TSESTree.CallExpression) => void;
+        AssignmentExpression: (node: TSESTree.AssignmentExpression) => void;
+        VariableDeclarator: (node: TSESTree.VariableDeclarator) => void;
+    } {
+        
+
+        const services = ESLintUtils.getParserServices(ruleContext);
+
+        const typeChecker = services.program.getTypeChecker();
+
+        //Registry of all functions that use Office API calls (regardless if CF or not)
+        let officeCallingFuncs = new Set<ts.Node>(); 
```

Comment:
Can you please define the type for this variable and variables throughout if possible